lib/did_you_mean/extra_features/key_error.rb

```diff
@@ -0,0 +1,25 @@
+module DidYouMean
+  module ExtraFeatures
+    module KeyErrorReciver
+      def fetch(name, *args, &block)
+        super
+      rescue KeyError => e
+        this = self
+        e.define_singleton_method(:receiver) { this }
+        e.define_singleton_method(:name) { name }
+        e.define_singleton_method(:args) { nil }
```

Comment:
Rather than dynamically defining methods within the method, it might be better to just use instance variables:

```ruby
module KeyErrorWithNameAndKeys
  def self.prepend(klass)
    klass.send(:attr, :keys, :name)
  end

  def fetch(name, *)
    super
  rescue KeyError => e
    e.instance_variable_set(:@keys, keys)
    e.instance_variable_set(:@name, name)

    raise e
  end
end

KeyError.prepend KeyErrorWithNameAndKeys
```